### PR TITLE
stripe integration

### LINKED ISSUE #37: research integration of stripe

### PR DESCRIPTION
**Pull request by petarblazevski**
- Handle the creation of the stripe link for connecting the company with the platform
- Handle stripe callback

### CODE REVIEW

File: libs/response-lib.js [hunk 1]
```diff
@@ -14,11 +14,22 @@ export function resourceNotFound(body) {
...
+  const url = `${location}?error=${error}&error_description=${errorMsg}`;
+  return redirect(url);
+}
+
+function buildResponse(statusCode, body, headers = {}) {
   return {
     statusCode,
     headers: {
-      'Access-Control-Allow-Origin': '*'
+      'Access-Control-Allow-Origin': '*',
```
Comment (IAfanasovMob) [inline libs/response-lib.js:30]: Just for my knowledge: I would expect `Access-Control-Allow-Origin` to be a GateWay setting. Is it correct to manage CORS from the lambda code?

File: stripe/connectCallback.js [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
+  let company;
+
+  try {
+    const response = await dynamoDbLib.call('scan', params);
+
+    if (response.Count === 0) {
+      const error = 'resource_not_found';
+      const errorMsg = 'The company does not exist in the DB';
+      return redirectWithError(process.env.APPLICATION_URL, error, errorMsg);
```
Comment (IAfanasovMob) [inline stripe/connectCallback.js:26]: should we log some info to track such failed requests?

Comment (IAfanasovMob) [reply stripe/connectCallback.js:26]: I suppose if AWS somehow logs all the requests with all the query params and other params, it would be good enough

**Review by IAfanasovMob — APPROVED** [review]
I don't know if we have enough logging to track failed stripe callback calls. It is probably luck of aws knowledge. Please feel free to merge if you are sure we have enough logging
Overall it is really good!

File: libs/response-lib.js (hunk 1 above)
Comment (petarblazevski) [reply libs/response-lib.js:30]: This is just so the lambda can communicate with API Gateway. The gateway will take care of this. I am not an expect :D, this is the recommended setting from the Serverless Framework when you have API Gateway and Cognito

**Comment by petarblazevski** [comment]
There are things we need to take care of, like more validations and logging, as you mentioned. I will update this PR with some `console.error`, so we can keep track in AWS.

We need to keep in mind, this is not a complete solution, just a working solution right now

**Review by IAfanasovMob — APPROVED** [review]
awesome!

**Pull request merged**